### PR TITLE
fix bug of loss nan

### PR DESCRIPTION
**Pull request by cuhk-hbsun**
Fix ctc_loss NaN bug by setting `zero_infinity=True, flatten=False` for ctc_loss

### CODE REVIEW

File: mmocr/models/textrecog/losses/ctc_loss.py [hunk 1]
```diff
@@ -54,6 +54,8 @@ def forward(self, outputs, targets_dict):
                 targets[idx, :valid_len] = tensor[:valid_len]
 
         target_lengths = targets_dict['target_lengths']
+        target_lengths = torch.clamp(
+            target_lengths, min=1, max=seq_len - 1).long()
```
Comment (jeffreykuang) [inline mmocr/models/textrecog/losses/ctc_loss.py:58]: How about max=seq_len instead of max=seq_len-1?

**Comment by jeffreykuang** [comment]
Since zero_infinity=True, flatten=False is the key for avoiding nan loss. Why not set these value by default in the ctc loss initialization?  Is there any side effect if doing so?   If let users to set them in cofig.  I think it will result in difficulty of usage.

File: mmocr/models/textrecog/losses/ctc_loss.py [hunk 2]
```diff
@@ -54,6 +59,16 @@ def forward(self, outputs, targets_dict):
                 targets[idx, :valid_len] = tensor[:valid_len]
 
         target_lengths = targets_dict['target_lengths']
+        target_lengths = torch.clamp(target_lengths, min=1, max=seq_len).long()
+
+        input_lengths = torch.full(
+            size=(bsz, ), fill_value=seq_len, dtype=torch.long)
+        if not self.flatten and valid_ratios is not None:
```
Comment (jeffreykuang) [inline mmocr/models/textrecog/losses/ctc_loss.py:66]: why self.flatten=False?  How about if it is true?

Comment (cuhk-hbsun) [reply mmocr/models/textrecog/losses/ctc_loss.py:66]: If flatten is True, `input_lengths` must be filled with the value `seq_len` to use cudnn.

File: mmocr/models/textrecog/losses/ctc_loss.py [hunk 3]
```diff
@@ -36,12 +38,15 @@ def __init__(self,
         self.ctc_loss = nn.CTCLoss(
             blank=blank, reduction=reduction, zero_infinity=zero_infinity)
 
-    def forward(self, outputs, targets_dict):
+    def forward(self, outputs, targets_dict, img_metas=None):
```
Comment (jeffreykuang) [inline mmocr/models/textrecog/losses/ctc_loss.py:41]: When is the img_metas is None and when is it not None? How to control it? Can we set it in config?

**Review by jeffreykuang — APPROVED** [review]
